src/main/java/graphql/execution/AbstractAsyncExecutionStrategy.java

```diff
@@ -22,11 +22,12 @@ public AbstractAsyncExecutionStrategy(DataFetcherExceptionHandler dataFetcherExc
     }
 
     protected BiConsumer<List<Object>, Throwable> handleResults(ExecutionContext executionContext, List<String> fieldNames, CompletableFuture<ExecutionResult> overallResult) {
-        return (List<Object> results, Throwable exception) -> executionContext.run(() -> {
+        return (List<Object> results, Throwable exception) -> executionContext.run(exception, () -> {
```

Comment:
whenever we call  `run` here we need to respect if an exception was thrown on the inside when we check if we are cancelled.

There is not point throwing a "cancelled" exception if we are already in exception